lib/index.js

```diff
@@ -179,32 +184,60 @@ export function configureUnifiedLanguageServer(
   async function processDocuments(textDocuments, alwaysStringify = false) {
     // LSP uses `file:` URLs (hrefs), `unified-engine` expects a paths.
     // `process.cwd()` does not add a final slash, but `file:` URLs often do.
-    const workspacesAsPaths = [...workspaces].map((d) =>
-      fileURLToPath(d.replace(/\/$/, ''))
-    )
+    const workspacesAsPaths = [...workspaces]
+      .map((d) => d.replace(/[/\\]?$/, ''))
+      // Sort the longest (closest to the file) first.
+      .sort((a, b) => b.length - a.length)
     /** @type {Map<string, Array<VFile>>} */
     const workspacePathToFiles = new Map()
 
-    if (workspacesAsPaths.length === 0) {
-      workspacesAsPaths.push(process.cwd())
-    }
+    await Promise.all(
+      textDocuments.map(async (textDocument) => {
+        /** @type {string | undefined} */
+        let cwd
+        if (workspaces.size === 0) {
+          cwd = await findUp(
+            async (dir) => {
+              const pkgExists = await pathExists(path.join(dir, 'package.json'))
+              if (pkgExists) {
+                return dir
+              }
 
-    for (const textDocument of textDocuments) {
-      const file = lspDocumentToVfile(textDocument)
-      const [cwd] = workspacesAsPaths
-        // Every workspace that includes the document.
-        .filter((d) => file.path.slice(0, d.length + 1) === d + path.sep)
-        // Sort the longest (closest to the file) first.
-        .sort((a, b) => b.length - a.length)
-
-      // This presumably should not occur: a file outside a workspace.
-      // So ignore the file.
-      /* c8 ignore next */
-      if (!cwd) continue
-
-      const files = workspacePathToFiles.get(cwd) || []
-      workspacePathToFiles.set(cwd, [...files, file])
-    }
+              const gitExists = await pathExists(path.join(dir, '.git'))
+              // Git doesn’t allow us to commit files in a .git folder.
```

Comment:
I don’t understand this comment?
You could write a `.git` file/folder in `test`, to test this?
Or: `// Should be fine` if we can’t be bothered to test this?

Reply:
You’re right, a test can just write the `.git` directory. This is now properly tested.